src/c++/vdo/base/repair.c

```diff
@@ -1611,8 +1611,37 @@ static int parse_journal_for_recovery(struct repair_completion *repair)
 	sequence_number_t i, head;
 	bool found_entries = false;
 	struct recovery_journal *journal = repair->completion.vdo->recovery_journal;
+	enum vdo_metadata_type expected_format;
 
 	head = min(repair->block_map_head, repair->slab_journal_head);
+	expected_format = get_recovery_journal_block_header(journal,
+							    repair->journal_data,
+							    head).metadata_type;
+	for (i = head; i <= repair->highest_tail; i++) {
+		struct recovery_block_header header;
```

Comment:
I woudl move the "struct recovery_block_header" declaration outside this loop. It's used in both loops so you'll only have to declare once that way. Also, having the intermeiate 'header' variable will make spelling less awkward when setting the expected_format (line 1617).

Reply:
Ah yes - had meant to do this and forgot.